adagrams/game.py

```diff
@@ -1,11 +1,149 @@
+}
+
+VALUE_SCORES = {
+'A': 1, 'E': 1, 'I': 1, 'O': 1, 'U': 1, 'L': 1, 'N': 1, 'R': 1, 'S': 1, 'T': 1,
+'D': 2, 'G': 2,
+'B': 3, 'C': 3, 'M': 3, 'P': 3,
+'F': 4, 'H': 4, 'V': 4, 'W': 4, 'Y': 4,
+'K': 5,
+'J': 8, 'X': 8,
+'Q':10, 'Z':10
+}
+
+
+#WAVE 1
 def draw_letters():
-    pass
+    #pass
+    #no parameters
+    #return an array of 10 strings
+    # array is a list, return list of 10 strings
+    drawn_letters = []
+    available_letters = []
+
+    #each string should contain exactly one letter
+    #these represent the hand of letters that the player has drawn
+    # LETTER_POOL is a dict - letter, num is key, value
+    #.items returns each item in a dict as tuples in a list
+    for letter, num in LETTER_POOL.items():
+        for i in range(num):
+            available_letters.append(letter)
+
+    # letters should be randomly drawn from a pool of letters
+    # letter pool should reflect the distribution of letters
+    # make sure the drawn letters match the rules of the letter pool
+    # while length of drawn letters from pool of letters is <= 10 is True
+    while len(drawn_letters) < 10:
+        # draw random letter from available_letters
+        # use import random to use random() which returns random float between 0 and 1
+        # use choice() returns a random element from the given sequence
+        randomly_draw_letter = random.choice(available_letters)
+        available_letters.remove(randomly_draw_letter)
+        # print random drawn letter to reflect the distribution of letters
+        #print(randomly_draw_letter)
+        drawn_letters.append(randomly_draw_letter)
+        #build a data sturcture for the letter pool
+    #there are only 2 available C letters
+        #draw_letters cannot return more than 2 C's
+    #there are 12 E's but only one Z
+        #should be 12 times as likely to draw a E as a Z
+    #invoking this function should NOT change the pool of letters
+        #imagine user returns their hand to the pool before drawing new letterss
+    return drawn_letters
+
+#terminal vs code wave one success! -- terminal wave one success!
 
+#WAVE 2
 def uses_available_letters(word, letter_bank):
-    pass
+    #pass
+    #nOT_cAsE_sEnsiTiVe = word.upper() -- could have made a variable
+    #letter bank describes an array of drawn letters in a hand
+    for letter in list(word.upper()):
+    # set is a dict unordered duplicates not allowed -- set would work if using with list, tuple or dict
+    #set didn't work was not iterating thru for loop
+    # had to turn set into a list
+    #check if an input word only uses characters that are within a hand of drawn cards
+    #word describes some input word, and is a string
+        if word.upper().count(letter) > letter_bank.count(letter):
+            return False
+    #return False if not; if there is a letter in input that is not present in the letter_bank
+    #return True if every letter in the input word is available in the letter_bank
+    return True
```

Comment:
Nicely done! Actually, your `uses_available_letters` solution is one of the most efficient and logical ones I've seen (essentially 4 lines long!!). Nice work.